sdk/servicebus/azure-servicebus/azure/servicebus/aio/management/_management_client_async.py

```diff
@@ -927,10 +945,8 @@ async def update_subscription(
         """
 
         _validate_entity_name_type(topic_name, display_name="topic_name")
-
-        subscription = create_properties_from_dict_if_needed(
-            subscription, SubscriptionProperties
-        )
+        # we should not mutate the input, making a copy first for update
+        subscription = deepcopy(create_properties_from_dict_if_needed(subscription, SubscriptionProperties))
```

Comment:
Is deepcopy actually returning an instance of `SubscriptionProperties` rather than a dict?